auto_flu/analysis.py

```diff
@@ -0,0 +1,95 @@
+import datetime
+import json
+import logging
+import os
+import shutil
+import subprocess
+
+
+def build_pipeline_command(config, pipeline):
+    """
+    Builds the pipeline command to be executed.
+
+    :param config: The config dictionary
+    :type config: dict
+    :param pipeline: The pipeline dictionary
+    :type pipeline: dict
+    :return: The pipeline command to be executed, as a list of strings
+    :rtype: list
+    """
+
+    pipeline_command = [
+            'nextflow',
+            '-log', pipeline['pipeline_parameters']['log_path'],
+            'run',
+            pipeline['pipeline_name'],
+            '-r', pipeline['pipeline_version'],
+            '-profile', 'conda',
+            '--cache', os.path.join(os.path.expanduser('~'), '.conda/envs'),
+            '-work-dir', pipeline['pipeline_parameters']['work_dir'],
+            '-with-report', pipeline['pipeline_parameters']['report_path'],
+            '-with-trace', pipeline['pipeline_parameters']['trace_path'],
+            '-with-timeline', pipeline['pipeline_parameters']['timeline_path'],
+    ]
+    pipeline['pipeline_parameters'].pop('log_path', None)
+    pipeline['pipeline_parameters'].pop('work_dir', None)
+    pipeline['pipeline_parameters'].pop('report_path', None)
+    pipeline['pipeline_parameters'].pop('trace_path', None)
+    pipeline['pipeline_parameters'].pop('timeline_path', None)
+
+    for flag, value in pipeline['pipeline_parameters'].items():
+        if value is None:
+            pipeline_command += ['--' + flag]
+        else:
+            pipeline_command += ['--' + flag, value]
+    
+    return pipeline_command
+
+
+def run_pipeline(config, pipeline, run):
+    """
+    Analyzes a run.
+
+    :param config: The config dictionary
+    :type config: dict
+    :param run: The run dictionary
+    :type run: dict
+    :param pipeline: The pipeline dictionary
+    :type pipeline: dict
+    :return: None
+    :rtype: None
+    """
+
+    analysis_tracking = {
+        "timestamp_analysis_start": datetime.datetime.now().isoformat()
+    }
+    pipeline_command = build_pipeline_command(config, pipeline)
+    pipeline_command_str = list(map(str, pipeline_command))
+
+    sequencing_run_id = run['sequencing_run_id']
+    
+    try:
+        logging.info(json.dumps({
+            "event_type": "analysis_started",
+            "sequencing_run_id": sequencing_run_id,
+            "pipeline_command": pipeline_command_str
+        }))
+        print(pipeline_command_str)
+        analysis_result = subprocess.run(pipeline_command_str, capture_output=True, check=True)
```

Comment:
This fails with a rather generic error: 

```
Traceback (most recent call last):
  File "/sharehome/john.palmer/git/auto-flu/auto_flu/analysis.py", line 79, in run_pipeline
    analysis_result = subprocess.run(pipeline_command_str, capture_output=True, check=True)
  File "/home/john.palmer/mamba/envs/test/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['nextflow', '-log', '/home/john.palmer/work/flu/results/test_analysis_results/240210_VH00502_167_AAF3VGCM5/fluviewer-nf-v0.3-output/240210_VH00502_167_AAF3VGCM5_fluviewer-nf_nextflow.log', 'run', 'BCCDC-PHL/fluviewer-nf', '-r', 'v0.3.0', '-profile', 'conda', '--cache', '/home/john.palmer/.conda/envs', '-work-dir', '/home/john.palmer/work/flu/work/work-240210_VH00502_167_AAF3VGCM5_fluviewer-nf_20240718095441', '-with-report', '/home/john.palmer/work/flu/results/test_analysis_results/240210_VH00502_167_AAF3VGCM5/fluviewer-nf-v0.3-output/240210_VH00502_167_AAF3VGCM5_fluviewer-nf_report.html', '-with-trace', '/home/john.palmer/work/flu/results/test_analysis_results/240210_VH00502_167_AAF3VGCM5/fluviewer-nf-v0.3-output/240210_VH00502_167_AAF3VGCM5_fluviewer-nf_trace.tsv', '-with-timeline', '/home/john.palmer/work/flu/results/test_analysis_results/240210_VH00502_167_AAF3VGCM5/fluviewer-nf-v0.3-output/240210_VH00502_167_AAF3VGCM5_fluviewer-nf_timeline.html', '--fastq_input', '/home/john.palmer/work/flu/fastq/test_fastq_symlinks/240210_VH00502_167_AAF3VGCM5', '--db', '/data/ref_databases/fluviewer/20230908/FluViewer_db.fa', '--outdir', '/home/john.palmer/work/flu/results/test_analysis_results/240210_VH00502_167_AAF3VGCM5/fluviewer-nf-v0.3-output']' returned non-zero exit status 1.
```
I copied that exact command to an interactive Python session, and it ran with no issues.